File: packages/typescript/temporal-yaml/examples/claims-review-composition/activities.ts
/**
 * Code-injected activities (the YAML+code authoring mode). `finalize` is a
 * code-DEFINED activity descriptor passed to `buildRuntime` via `extraActivities`
 * instead of declared inline in the YAML — the TS analogue of Python's
 * `activities.modules`. The referenced sub-workflows and the parent's other two
 * activities stay pure-YAML — both authoring modes composed in one project.
 */

import { defineActivity } from "@typeflux/temporal";
import { z } from "zod";

export const finalize = defineActivity({
  name: "finalize",
  prompt: { name: "finalize-review" },
  input: z.object({ summary: z.string(), escalate: z.boolean() }),
  output: z.object({ decision: z.string() }),
});

export const extraActivities = { finalize };

File: packages/python/examples/claims_review_composition/activities.py
"""Code-injected activities for the parent workflow (the YAML+code authoring mode).

The parent ``claims_review.yaml`` declares ``activities.modules: [activities]`` instead
of inline ``activities.definitions``, so these ``AIActivity`` objects are imported and
composed with the graph features (parallel / when / sub-workflows / gates). The
sub-workflow children (``claim_triage`` / ``escalation_review``) stay pure-YAML — one
project exercises BOTH authoring modes composed together (#55 scope addition A).
"""

from __future__ import annotations

from examples.claims_review_composition.schemas import (
    Acknowledgement,
    ClaimBatch,
    Consolidated,
    IntakeFanout,
    ReviewPacket,
)
from typeflux.core import AIActivity, PromptRef

acknowledge = AIActivity(
    name="acknowledge",
    input_type=ClaimBatch,
    output_type=Acknowledgement,
    prompt_ref=PromptRef("acknowledge-intake"),
)
consolidate = AIActivity(
    name="consolidate",
    input_type=IntakeFanout,
    output_type=Consolidated,
    prompt_ref=PromptRef("consolidate-intake"),
)
finalize = AIActivity(
    name="finalize",
    input_type=Consolidated,
    output_type=ReviewPacket,
    prompt_ref=PromptRef("finalize-review"),
)

ALL_ACTIVITIES = (acknowledge, consolidate, finalize)
